SmartEVSE2.X/modbus.c

```diff
@@ -475,49 +483,65 @@ signed long receiveMeasurement(unsigned char *buf, unsigned char pos, unsigned c
 
 /**
  * Send Energy measurement request over modbus
- * 
+ *
  * @param unsigned char Meter
  * @param unsigned char Address
  */
 void requestEnergyMeasurement(unsigned char Meter, unsigned char Address) {
-    ModbusReadInputRequest(Address, EMConfig[Meter].Function, EMConfig[Meter].ERegister, 2);
+    switch (Meter) {
+        case EM_SOLAREDGE:
+            // Read 3 Current values + FIXME: scaling factor // 32bit values
+            ModbusReadInputRequest(Address, EMConfig[Meter].Function, EMConfig[Meter].ERegister, (EMConfig[Meter].DataType == MB_DATATYPE_INT16 ? 3 : 6) + 1);
```

Comment:
On EM_SOLAREDGE DataType is always MB_DATATYPE_INT16, so maybe safe the extra if clause and only use value "3"